src/runtime_src/core/tools/common/SubCmd.h

```diff
@@ -20,22 +20,56 @@
 // Please keep eternal include file dependencies to a minimum
 #include <vector>
 #include <string>
-#include <map>
+#include <boost/program_options.hpp>
   
-typedef int (*t_subcommand)(const std::vector<std::string> &);
-  
-typedef struct {
-  std::string sSubCmd;
-  std::string sDescription;
-  t_subcommand callBackFunction;
-  bool isHidden;
-} SubCmdEntry;
+class SubCmd {
+ public:
+   typedef std::vector<std::string> SubCmdOptions;
+
+ public:
+   virtual void execute(const SubCmdOptions &_options) const = 0;
+
+ public:
+   const std::string &getName() const { return m_subCmdName; };
+   const std::string &getShortDescription() const { return m_shortDescription; };
+   bool isHidden() const { return m_isHidden; };
+   bool isDeprecated() const { return m_isDeprecated; };
+   bool isPreliminary() const { return m_isPreliminary; };
 
-typedef std::map<const std::string, SubCmdEntry> SubCmdTable;
+ public:
+   void setExecutableName(const std::string & _name) { m_executableName = _name; };
+   const std::string & getExectuableName() const {return m_executableName; };
 
-unsigned int register_subcommand(const std::string &_subCommandName, const std::string &_description, t_subcommand _pSubCommand, bool _isHidden = false);
-const SubCmdEntry *getSubCmdEntry(const std::string &_sSubCmdName);
-const SubCmdTable &getSubCmdsTable();
+ public:
+   virtual ~SubCmd() {};
 
+ // Child class Helper methods
+ protected:
+  SubCmd(const std::string _name, const std::string _shortDescription);

```

Comment:
Since you are not moving these value arguments in the ctor, please use const std::string&

Reply:
done